cocos/platform/CCFileUtils.cpp

```diff
@@ -846,6 +846,7 @@ void FileUtils::setWritablePath(const std::string& writablePath)
 void FileUtils::setDefaultResourceRootPath(const std::string& path)
 {
     _defaultResRootPath = path;
+    setSearchPaths(_searchPathArray);
```

Comment:
Yep, good catch.

But i think this line
https://github.com/kpkhxlgy0/cocos2d-x/blob/e87207462912eef8235dfcab568eb4405a599002/cocos/platform/CCFileUtils.cpp#L857

will make the `searchPaths` parameter cleared.

We need to make a copy of `searchPaths` parameter before `clear`.